.teamcity/src/main/kotlin/configurations/FlakyTestQuarantine.kt

```diff
@@ -31,11 +32,23 @@ class FlakyTestQuarantine(model: CIBuildModel, stage: Stage, os: Os, arch: Arch
         )
     }.flatMap { it.functionalTests }.filter { it.os == os }
 
+    if (os == Os.LINUX) {
+        steps {
+            script {
+                // Because we exclude tests in `distributions-integ-tests` below, `@Flaky` won't work in that subproject.
+                // Here we check the existence of `@Flaky` annotation to make sure nobody use that annotation in `distributions-integ-tests` subproject.
+                name = "MAKE_SURE_NO_@FLAKY_IN_DISTRIBUTIONS_INTEG_TESTS"
+                executionMode = BuildStep.ExecutionMode.ALWAYS
+                scriptContent = "! grep 'org.gradle.test.fixtures.Flaky' -r subprojects/distributions-integ-tests/src"
```

Comment:
I'd prefer having that somewhere as part of the build. One way would be to add some check in the base class for those tests for example.

Did you try out if this script fails CI when there is an `@Flaky` annotation somewhere on those tests?

Reply:
I wonder how. By inspecting the annotation of each method via reflection? That look weird. Also, this doesn't prevent people from using `@Flaky` in new classes.

This solution is very cheap and I've tested it. Also, I like to put it here because `-x distributions-integ-tests:platformTest` is also defined here.

Reply:
We can leave it like it is for now. My intention to move it to the build would be that whoever makes the change gets feedback earlier instead of recently in the ready for release job.